renderdoc/driver/shaders/dxbc/dxbc_debug.cpp

```diff
@@ -573,6 +585,25 @@ bool State::Finished() const
   return dxbc && (done || nextInstruction >= (int)dxbc->GetNumInstructions());
 }
 
+void State::AssignValue(ShaderVariable &dst, uint32_t dstIndex, const ShaderVariable &src,
+                        uint32_t srcIndex)
+{
+  if(src.type == eVar_Float)
+  {
+    float ft = src.value.fv[srcIndex];
+    if(isinf(ft) || isnan(ft))
+      flags |= eShaderDebugStateFlags_GeneratedNanOrInf;
+  }
+  else if(src.type == eVar_Double)
+  {
+    double dt = src.value.dv[srcIndex];
+    if(isinf(dt) || isnan(dt))
+      flags |= eShaderDebugStateFlags_GeneratedNanOrInf;
+  }
```

Comment:
Yeh unfortunately there's nothing in the bytecode itself that describes them, we'd have to generate such a table ourselves and like you say it'd probably get bloated for little reason.

I was thinking aloud, since float exceptions on CPU only happen when those float values are generated not moved around, but it probably happens pretty similarly - you mostly care about the initial nan. You don't care how it propagates from there.